python/triton/language/core.py

```diff
@@ -887,7 +888,7 @@ def store(pointer, value, mask=None, cache_modifier="", eviction_policy="", _bui
     """
     # value can be constexpr
     value = _to_tensor(value, _builder)
-    if _constexpr_to_value(mask) is not None:
+    if mask is not None and _constexpr_to_value(mask) is not None:
```

Comment:
I don't really understand the point of this additional check, since `_constexpr_to_value` returns `None` when its input is `None`

Reply:
That's correct, this has no impact on functionality, but it silences a type checker warning that `_to_tensor` does not take `None`. The type checker does not evaluate `_constexpr_to_value` to determine this. Making it explicit silences the warning.

Reply:
Couldn't we silence the typechecker with some return type hint on `_constexpr_to_value` ? In general I'm pretty hostile to creating redundancy in the code for the sake of silencing the typechecker.